odl/discr/lp_discr.py

```diff
@@ -387,14 +385,9 @@ def __repr__(self):
             use_uniform = False
 
         if use_uniform:
-            if isinstance(self.dspace, Fn):
-                impl = 'numpy'
-                default_dtype = np.float64
-            elif isinstance(self.dspace, CudaFn):
-                impl = 'cuda'
-                default_dtype = np.float32
-            else:  # This should never happen
-                raise RuntimeError('unable to determine data space impl')
+            impl = self.dspace.impl
+            default_dtype = self.dspace.default_dtype(RealNumbers())
```

Comment:
We can add `default_order` to an `NtuplesBase` type space if we want, so that info would come in "automatically" by the plug-in. Seems a bit meaningless for one-dimensional storage, though, it becomes meaningful only in combination with `DiscretizedSet`. One we have multi-dimensional storage (*knock knock) this one becomes more interesting.